test/plots/rectanglePlotTests.ts

```diff
@@ -2,511 +2,486 @@
-      assert.strictEqual(cellAU.attr("width"), "200", "cell 'AU' width is correct");
-      assert.strictEqual(cellAU.attr("x"), "0", "cell 'AU' x coord is correct");
-      assert.strictEqual(cellAU.attr("y"), "0", "cell 'AU' y coord is correct");
-      assert.strictEqual(cellAU.attr("fill"), "#000000", "cell 'AU' color is correct");
-
-      assert.strictEqual(cellBU.attr("height"), "100", "cell 'BU' height is correct");
-      assert.strictEqual(cellBU.attr("width"), "200", "cell 'BU' width is correct");
-      assert.strictEqual(cellBU.attr("x"), "200", "cell 'BU' x coord is correct");
-      assert.strictEqual(cellBU.attr("y"), "0", "cell 'BU' y coord is correct");
-      assert.strictEqual(cellBU.attr("fill"), "#212121", "cell 'BU' color is correct");
-
-      assert.strictEqual(cellAV.attr("height"), "100", "cell 'AV' height is correct");
-      assert.strictEqual(cellAV.attr("width"), "200", "cell 'AV' width is correct");
-      assert.strictEqual(cellAV.attr("x"), "0", "cell 'AV' x coord is correct");
-      assert.strictEqual(cellAV.attr("y"), "100", "cell 'AV' y coord is correct");
-      assert.strictEqual(cellAV.attr("fill"), "#ffffff", "cell 'AV' color is correct");
-
-      assert.strictEqual(cellBV.attr("height"), "100", "cell 'BV' height is correct");
-      assert.strictEqual(cellBV.attr("width"), "200", "cell 'BV' width is correct");
-      assert.strictEqual(cellBV.attr("x"), "200", "cell 'BV' x coord is correct");
-      assert.strictEqual(cellBV.attr("y"), "100", "cell 'BV' y coord is correct");
-      assert.strictEqual(cellBV.attr("fill"), "#777777", "cell 'BV' color is correct");
-    };
-
-    it("renders correctly", () => {
-      let xScale = new Plottable.Scales.Category();
-      let yScale = new Plottable.Scales.Category();
-      let colorScale = new Plottable.Scales.InterpolatedColor();
-      colorScale.range(["black", "white"]);
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let gridPlot = new Plottable.Plots.Rectangle();
-      gridPlot.addDataset(new Plottable.Dataset(DATA))
-              .attr("fill", (d) => d.magnitude, colorScale);
-      gridPlot.x((d: any) => d.x, xScale)
-              .y((d: any) => d.y, yScale);
-      gridPlot.renderTo(svg);
-      VERIFY_CELLS((<any> gridPlot)._renderArea.selectAll("rect")[0]);
-      svg.remove();
-    });
+      it("renders correctly when data is set after construction", () => {
+        let dataset = new Plottable.Dataset();
+        let plot = new Plottable.Plots.Rectangle();
+        plot.addDataset(dataset);
+        plot.attr("fill", (d) => d.magnitude, colorScale);
+        plot.x((d: any) => d.x, xScale);
+        plot.y((d: any) => d.y, yScale);
+        plot.renderTo(svg);
+        dataset.data(data);
+        verifyCells(plot.content().selectAll("rect")[0]);
+        svg.remove();
+      });
+
+      it("renders correctly when there isn't data for every spot", () => {
```

Comment:
That would result in duplication. Maybe I'll give it a break until second pass, when I see how I deal with this in other scenarios.